internal/driver/flamegraph.go

```diff
@@ -0,0 +1,363 @@
+			.transitionEase(d3.easeCubic)
+			.sort(true)
+			.title("")
+			.label(label);
+
+		var tip = d3.tip()
+			.direction("s")
+			.offset([8, 0])
+			.attr('class', 'd3-flame-graph-tip')
+			{{if eq .Unit "nanoseconds"}}
+			.html(function(d) { return "name: " + d.data.n + ", value: " + d3.format(".5f")(d.data.v / 1000000000) + " seconds"; });
+			{{else}}
+			.html(function(d) { return "name: " + d.data.n + ", value: " + d.data.v; });
+			{{end}}
+
+		flameGraph.tooltip(tip);
+
+		d3.select("#chart")
+			.datum(data)
+			.call(flameGraph);
+
+		document.getElementById("form").addEventListener("submit", function(event){
+			event.preventDefault();
+			search();
+		});
+
+		function search() {
+			var term = document.getElementById("term").value;
+			flameGraph.search(term);
+		}
+		document.getElementById("search").addEventListener("click", search);
+
+		function clear() {
+			document.getElementById('term').value = '';
+			flameGraph.clear();
+		}
+		document.getElementById("clear").addEventListener("click", clear);
+
+		function resetZoom() {
+			flameGraph.resetZoom();
+		}
+		document.getElementById("reset").addEventListener("click", resetZoom);
+		
+		window.addEventListener("resize", function() {
+			var width = document.getElementById("chart").clientWidth;
+			var graphs = document.getElementsByClassName("d3-flame-graph");
+			if (graphs.length > 0) {
+				graphs[0].setAttribute("width", width);
+			}
+			flameGraph.width(width);
+			flameGraph.resetZoom();
+		}, true);
+	</script>
+  </body>
+</html>`))
+
+type flameGraphNode struct {
+	Name     string
+	Value    int64
+	Children map[string]*flameGraphNode
+}
+
+func (n *flameGraphNode) add(stackPtr *[]string, index int, value int64) {
+	n.Value += value
+	if index >= 0 {
+		head := (*stackPtr)[index]
+		childPtr, ok := n.Children[head]
+		if !ok {
+			childPtr = &(flameGraphNode{head, 0, make(map[string]*flameGraphNode)})
+			n.Children[head] = childPtr
+		}
+		childPtr.add(stackPtr, index-1, value)
+	}
+}
+
+func (n *flameGraphNode) MarshalJSON() ([]byte, error) {
+	v := make([]flameGraphNode, 0, len(n.Children))
+	for _, value := range n.Children {
+		v = append(v, *value)
+	}
+
+	return json.Marshal(&struct {
+		Name     string           `json:"n"`
+		Value    int64            `json:"v"`
+		Children []flameGraphNode `json:"c"`
+	}{
+		Name:     n.Name,
+		Value:    n.Value,
+		Children: v,
+	})
+}
+
+// flamegraph generates a web page containing a flamegraph.
+func (ui *webInterface) flamegraph(w http.ResponseWriter, req *http.Request) {
+	if req.URL.Path != "/flamegraph" {
+		http.NotFound(w, req)
+		return
+	}
+
+	// Capture any error messages generated while generating a report.
+	catcher := &errorCatcher{UI: ui.options.UI}
+	options := *ui.options
+	options.UI = catcher
+
+	// Get sample_index from variables
+	si := pprofVariables["sample_index"].value
+
+	// Get query parameters
+	t := req.URL.Query().Get("t")
+
+	// Defaulting to first SampleType in profile
+	index := 0
```

Comment:
Should default to the last index, see https://github.com/google/pprof/blob/d19a98b9bcec464f178b20da0f936cef7e15462e/proto/profile.proto#L76. You can just call SampleIndexByName, it does the right thing for the empty string value.

Reply:
Done.